deepcell_label_utils/utils.py

```diff
@@ -93,14 +99,100 @@ def int_coords(x):
+            zf = zip_obj
+        else:
+            print("No zip specified...")
+            return
+        data = zf.read('X.ome.tiff')
+        bytes_io = io.BytesIO(data)
+        X_init = TiffFile(bytes_io).asarray(squeeze=False)
+        self.X_ome = self.reshape_DCL(X_init)  # Reshape from _CTYX_ to BTZYXC
+        self.X = self.to_TYXC(self.X_ome)      # Reshape to TYXC for SLC
+
+        data = zf.read('y.ome.tiff')
+        bytes_io = io.BytesIO(data)
+        y_init = TiffFile(bytes_io).asarray(squeeze=False)
+        self.y_ome = self.reshape_DCL(y_init)  # Reshape from _CTYX_ to BTZYXC
+        self.y = self.to_TYXC(self.y_ome)      # Reshape to TYXC for SLC
+
+        data = zf.read('cells.json')
+        cells = json.loads(data.decode('utf-8'))
+        self.cells = cells
+        self.segments = self.segments_to_df(self.cells)  # To df for SLC
+
+        data = zf.read('divisions.json')
+        divisions = json.loads(data.decode('utf-8'))
+        self.divisions = divisions
+
+        zf.close()
+
+    def reshape_DCL(self, arr):
+        """
+        Rearrange the axes of the DCL output tiffs to the lab
+        standard BTZYXC.
+
+        NOTE:
+            DCL currently puts the time axis in the Z dimension,
+            so the tiff shape is called CZYX but actually CTYX.
+        """
+        rearranged = np.transpose(arr, axes=[0, 2, 5, 3, 4, 1])
+        return rearranged
+
+    def to_TYXC(self, arr):
+        """
+        Convert a BTZYCX tiff array to TYXC. This allows for use
+        with the SpatialLabelConverter.
+
+        Raises:
+            ValueError: B and/or Z dimensions do not have length 1
+        """
+        try:
+            reshaped = np.squeeze(arr, axis=(0, 2))
+            if len(reshaped.shape) != 4:
+                raise ValueError
+            return reshaped
+        except ValueError:
+            print("Could not squeeze to TYXC, check dimensions.")
+
+    def segments_to_df(self, segments):
+        """
+        Convert segments list to Pandas dataframe. This allows
+        for use with the SpatialLabelConverter.
+        """
+        df = pd.DataFrame(segments)
+        return df
+
+
 class SpatialLabelConverter(object):
     """Label converter class for DeepCell Label label format. Converts
     DCL labels into binary masks, centroids, bboxes, and polygons"""
 
-    def __init__(self, X, y, segments, test_no_poly=False):
+    def __init__(self, X=None, y=None, segments=None,
+                 *, zip_path=None, zip_obj=None, test_no_poly=False):
         self.X = X
         self.y = y
         self.segments = segments
+        if zip_path is not None:
+            DCL = DCLZipConverter(zip_path=zip_path)
+            self.X = DCL.X
+            self.y = DCL.y
+            self.segments = DCL.segments
+        elif zip_obj is not None:
+            DCL = DCLZipConverter(zip_obj=zip_obj)
+            self.X = DCL.X
+            self.y = DCL.y
+            self.segments = DCL.segments
```

Comment:
Suggestion to clean this up and just rely on the logic that is build into the zip converter. Pass both zip_path and zip_obj into the DCLZipConverter. If neither variable is defined the error will be raised inside of zip converter.